### PR TITLE
split end to end tests client/worker

### PR DESCRIPTION
**Pull request by ddubuc**
fix https://github.com/aneoconsulting/ArmoniK/issues/631

### CODE REVIEW

File: Common/src/Common/Ext/EnumExt.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace ArmoniK.DevelopmentKit.Common.Ext;
+
+/// <summary>
+/// </summary>
+public abstract class EnumExt
+{
+  /// <summary>
+  /// </summary>
+  /// <param name="value"></param>
```
Comment (aneojgurhem) [inline Common/src/Common/Ext/EnumExt.cs:35]: The doc is missing here

File: Common/src/Common/Ext/IEnumerableExt.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Collections.Generic
```
Comment (aneojgurhem) [inline Common/src/Common/Ext/IEnumerableExt.cs:31]: This breaks the fact that git would have been able to tell that this file was moved

**Review by alouvion-aneo — APPROVED** [review]

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Program.cs [hunk 3]
```diff
@@ -73,14 +73,12 @@ private static void Main(string[] args)
...
-    clientsContainers = args is
-                        {
-                          Length: > 0,
-                        }
-                          ? RetrieveSpecificTests(args)
-                          : RetrieveClientTests();
+    var clientsContainers = args is
+                            {
+                              Length: > 0,
+                            }
```
Comment (dbrasseur-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Program.cs:79]:
```suggestion
    var clientsContainers = args.Length > 0
```

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckGridServer/HeavyPayloadGridServerClient.cs [hunk 4]
```diff
@@ -25,21 +25,29 @@
 using System.Linq;
 
 using ArmoniK.Api.gRPC.V1;
-using ArmoniK.DevelopmentKit.Client.GridServer;
+using ArmoniK.DevelopmentKit.Client.Unified.Exceptions;
+using ArmoniK.DevelopmentKit.Client.Unified.Factory;
+using ArmoniK.DevelopmentKit.Client.Unified.Services;
+using ArmoniK.DevelopmentKit.Client.Unified.Services.Submitter;
```
Comment (dbrasseur-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckGridServer/HeavyPayloadGridServerClient.cs:31]: So the test is not "GridServer" anymore then ? Time for a rename ?

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckUnifiedApi/SimpleUnifiedAPITestClient.cs [hunk 5]
```diff
@@ -243,7 +244,7 @@ private void ClientStartup2(Service sessionService)
     var tasksRandom = sessionService.Submit("RandomTaskError",
                                             Enumerable.Range(1,
                                                              wantedCount)
-                                                      .Select(_ => ParamsHelper(25)),
+                                                      .Select(_ => ParamsHelper(90)),
```
Comment (dbrasseur-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckUnifiedApi/SimpleUnifiedAPITestClient.cs:247]: Careful that it doesn't lengthen the test for too long. With the pod killing itself after a few retries

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckGridServer/SimpleGridServerTest.cs [hunk 6]
```diff
@@ -98,7 +79,7 @@ public static double ComputeReduceCube(byte[] inputs)
                    .ToArray();
   }
 
-  public double[] RandomTaskError(double percentageOfFailure = 25)
+  public double[] RandomTaskError(double percentageOfFailure = 0.25)
```
Comment (dbrasseur-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckGridServer/SimpleGridServerTest.cs:82]:
```suggestion
  public double[] RandomTaskError(double percentageOfFailure = 25)
```

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs [hunk 7]
```diff
@@ -99,12 +80,12 @@ public static double ComputeReduceCube(byte[] inputs)
...
   }
 
-  public double[] RandomTaskError(double percentageOfFailure = 25)
+  public double[] RandomTaskError(double percentageOfFailure = 0.25)
   {
     var randNum = rd.NextDouble();
     if (randNum < percentageOfFailure / 100)
     {
-      throw new GridServerException("An expected failure in this random call");
+      throw new WorkerApiException("An expected failure in this random call");
```
Comment (dbrasseur-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs:88]: It's not linked to a WorkerApi failure, but a worker side exception. Should change the exception type here

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckGridServer/SimpleGridServerTest.cs (hunk 6 above)
Comment (dbrasseur-aneo) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckGridServer/SimpleGridServerTest.cs:82]: You divide by 100 after

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

**Review by dbrasseur-aneo — APPROVED** [review]

**Pull request merged**